src/libraries/Microsoft.Extensions.Configuration.Binder/gen/TypeSpec.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    internal record TypeSpec
+    {
+        public TypeSpec(ITypeSymbol type)
```

Comment:
Am I right that none of the "spec" types actually store the `ITypeSymbol` and only extract the data that they need from the symbols at construction?

Reply:
Yes.